test-integracion

PatricioPlem opened the pull request:
Integracion de la logica de mapa, ubicacion y viajes con todas las demas clases. El juego ya es funcional!

Commit:
Creado un mapa y ubicaciones basicos que permiten entender la logica de los viajes entre ubicaciones

Commit:
Minor changes

PatoPlem committed:
Merge remote-tracking branch 'origin/main' into mapa

PatoPlem committed:
"base para manejar los viajes. Falta pensar como se inician las peleas (sin llenar todo de ifs)"

lucas-abalos committed:
Merge pull request #5 from lucas-abalos/mapa

mergeo mapa con test-integracion

PatoPlem committed:
Integracion de todas las clases del juego para que sea funcional de principio
a fin

PatoPlem committed:
minor changes

i-colombo commented on El Desafío del Reino Encantado/src/reino/main/Main.java:
```diff
@@ -0,0 +1,93 @@
+                }
+                //si elige viaje, usar logica de viaje
+
+                List<Criatura> enemigos = mapa.getActual().getEnemigo();
```
Toda esta logica de pelea deberia suceder en el dominio y no aca afuera en el Main

i-colombo commented on El Desafío del Reino Encantado/src/reino/main/Main.java:
```diff
@@ -0,0 +1,93 @@
+                //ofrecer opciones
+                if (mapa.getActual().getEnemigo() == null) {
+                    menuNeutral(scanner, jugador);
+                    mapa.viajar();
```
Falta modelar el Juego (fachada) y pedirle las cosas como viajar, crear Jugador, etc

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Guerrero.java:
- No utilizar instanceOf (mala practica de programacion)
- En cambio se puede directamente delegar en el Peronaje "defensor.fuisteGolpeadoPorGuerrero()" en donde en la clase Troll se define lo que se necesite

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Jugador.java:
```diff
@@ -32,6 +30,7 @@ public void agregarRecompensa(Recompensa recompensa) {
+        System.out.println("Conseguiste la siguiente recompensa: " + recompensa);
     }
 
     public void intercambiarRecompensas() {
```
- Supongo que este metodo lo que intentan hacer es reclamar una lista de recompensas
- Por que guardan esas recompensas temporales en un atributo?
- Por que no recibir la lista de recompensas como parametro de este metodo directamente?

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Jugador.java:
```diff
@@ -66,30 +71,41 @@ public void intercambiarRecompensas() {
 
-    public boolean mejorarAtaqueHeroe(int cantidad) {
+    public boolean mejorarAtaqueHeroe(int cantidad, boolean isRecompensa) {
+        if (isRecompensa) {
```
Las dos partes del if hacen lo mismo (nivelAtaque+cantidad). Por que no dejar solo esto como objetivo del metodo y que el resto de validaciones lo hagan otros metodos?

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Jugador.java:
```diff
@@ -66,30 +71,41 @@ public void intercambiarRecompensas() {
     }
 
-    public boolean mejorarDefensaHeroe(int cantidad) {
+    public boolean mejorarDefensaHeroe(int cantidad, boolean isRecompensa) {
```
Mismo comentario que en el metodo anterior. Baja cohesion del metodo. El nombre del metodo es mejorarDefensa, entonces que mejore la defensa. Todo el resto no es parte del objetivo del metodo

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Mapa.java:
```diff
@@ -1,11 +1,89 @@
+    public void viajar(){
+        System.out.println("A donde quieres viajar?");
+        this.actual.verViajes();
+        Scanner myObj = new Scanner(System.in);
```
El modelo NO puede contener interfaz de usuario!! No utilizar scanners dentro del modelo

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Pelea.java:
```diff
@@ -9,7 +9,7 @@ public Pelea(Heroe heroe, Criatura criatura) {
     }
 
-    public boolean iniciarPelea() {
+    public void iniciarPelea() {
```
Por que necesitan guardar el Heroe y la Criatura como atributos? Por que no recibirlos como parametros de este metodo?

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Pelea.java:
```diff
@@ -24,12 +24,11 @@ public boolean iniciarPelea() {
-        return heroe.getVida() > 0; // Si gano el heroe devuelvo True y sigue el curso del juego, si es false se termina la partida
     }
 
     private boolean turnoHeroe(int cantGolpes) {
```
sobre el metodo `atacar(...)` por que toda la logica esta ahi? No se puede delegar al Heroe y a la Criatura para que hablen entre ellos para resolver la pelea?

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Ubicacion.java:
```diff
@@ -1,20 +1,57 @@
-    public boolean tieneCriatura(){
-        return criatura != null;
+
+    public Ubicacion Viajar(int destino){
```
metodos van en minuscula

i-colombo commented on El Desafío del Reino Encantado/src/reino/modelo/Ubicacion.java:
```diff
@@ -1,20 +1,57 @@
-        return criatura != null;
+
+    public Ubicacion Viajar(int destino){
+        return this.adyacentes.get(destino-1);
```
estaria bueno encontrar una manera mas bonita de viajar, como buscar el destino por Id o por nombre en lugar de posición

i-colombo commented on Mapa.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class Mapa {
```
por que esta clase esta repetida y fuera del package de modelo?

i-colombo commented on Ubicacion.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.List;
+import java.util.ArrayList;
+
+public class Ubicacion {
```
Esto tambien esta repetido

lucas-abalos committed:
Correcciones en diagrama UML

Correcciones en diagrama UML luego de feedback de primer entrega. Se agrega la capa fachada.
